tests/shared/utils/test_io.py

```diff
@@ -620,3 +623,22 @@ def test_deep_container_fingerprint_can_use_instance_fingerprint():
     m1 = np.asarray([[0.5, 3.1, 3.0], [1.1, 1.2, 1.3], [4.7, 0.3, 2.7]])
     f = Features(m1, "sentence", "text", "CountVectorsFeaturizer")
     assert rasa.shared.utils.io.deep_container_fingerprint(f) == f.fingerprint()
+
+
+# Can't manipulate the STDOUT file descriptor
+# in pytest (it's a io.StringIO instance not a file)
+def test_handle_print_blocking(monkeypatch: MonkeyPatch):
+    def mock_print(*args, **kwargs):
+        raise BlockingIOError()
+
+    mock = Mock()
+    monkeypatch.setattr(builtins, "print", mock_print)
+    monkeypatch.setattr(rasa.shared.utils.io, "handle_print_blocking", mock)
```

Comment:
If we're mocking the function we want to test, we're not actually testing it, but I'd suggest to at least assert that mock was called with the desired output (in addition to the `call_count` assertion).
Also pytest has a good fixture for capturing stdout: `capsys` (https://docs.pytest.org/en/7.1.x/how-to/capture-stdout-stderr.html), have you tried with that too?

Reply:
Tried with `capsys` but it captures only the text output to stdout/stderr and not the system calls to lock/unlock stdout. After trying with `capsys` and `capfd` I decided to mock it but there was no good way (that I know of) to assert that lock/unlock happened so I was left with this...